umh-core/cmd/main.go

```diff
@@ -53,13 +54,61 @@ func main() {
 
 	// Load the config
 	configManager := config.NewFileConfigManager()
-	config, err := configManager.GetConfig(ctx, 0)
+	configData, err := configManager.GetConfig(ctx, 0)
 	if err != nil {
 		sentry.ReportIssuef(sentry.IssueTypeFatal, log, "Failed to load config: %s", err)
 	}
 
+	// Extract environment variables and apply them to the config
+	configModified := false
+
+	// Extract AUTH_TOKEN from environment if present
+	authToken, err := env.GetAsString("AUTH_TOKEN", false, "")
+	if err == nil && authToken != "" {
+		log.Info("Using AUTH_TOKEN from environment variable")
+		configData.Agent.CommunicatorConfig.AuthToken = authToken
+		configModified = true
+	}
+
+	// Extract API_URL from environment if present
+	apiURL, err := env.GetAsString("API_URL", false, "")
+	if err == nil && apiURL != "" {
+		log.Info("Using API_URL from environment variable:", zap.String("url", apiURL))
+		configData.Agent.CommunicatorConfig.APIURL = apiURL
+		configModified = true
+	}
+
+	// Extract RELEASE_CHANNEL from environment if present
+	releaseChannel, err := env.GetAsString("RELEASE_CHANNEL", false, "")
+	if err == nil && releaseChannel != "" {
+		log.Info("Using RELEASE_CHANNEL from environment variable:", zap.String("channel", releaseChannel))
+		configData.Agent.ReleaseChannel = config.ReleaseChannel(releaseChannel)
+		configModified = true
+	}
+
+	// Extract multiple location environment variables (LOCATION_0 through LOCATION_6)
+	locations := make(map[int]string)
+	for i := 0; i <= 6; i++ {
+		locationKey := fmt.Sprintf("LOCATION_%d", i)
```

Comment:
what hsould happen if the location is empty? maybe one default value? "please-set-location"